subsys/net/lib/dns/dns_sd.c

```diff
@@ -822,6 +815,94 @@ int dns_sd_handle_ptr_query(const struct dns_sd_rec *inst,
 	return offset;
 }
 
+int dns_sd_handle_service_type_enum(const struct dns_sd_rec *inst,
+				    const struct in_addr *addr4, const struct in6_addr *addr6,
+				    uint8_t *buf, uint16_t buf_size)
+{
+	static const char query[] = { "\x09_services\x07_dns-sd\x04_udp\x05local" };
+	/* offset of '.local' in the above */
+	uint16_t domain_offset = DNS_SD_PTR_MASK | 35;
+	uint16_t proto;
+	int name_size;
+	uint16_t service_size;
+	uint16_t offset = sizeof(struct dns_header);
+	struct dns_rr *rr;
+	struct dns_header *rsp = (struct dns_header *)buf;
+
+	memset(rsp, 0, sizeof(*rsp));
```

Comment:
I think we should make sure that `sizeof(*rsp) <= buf_size` before calling `memset`, so something like
```
if (sizeof(*rsp) > (size_t)buf_size) {
     return -EINVAL;
}
```